packages-exp/auth-exp/src/core/user/token_manager.ts

```diff
@@ -147,21 +158,33 @@ export class StsTokenManager {
       );
       manager.expirationTime = expirationTime;
     }
+    if (isPassthroughMode) {
```

Comment:
is this necessary or can we pull the state from checking refreshToken?

Reply:
Yeah, this goes back to the question of "is the absence of a refresh token a sufficient indicator that we're in passthrough mode" and unfortunately, it isn't. Ties into one of the discussion threads from an earlier commit

Reply:
This if-clause will not handle `false`. Is this expected? If not, try `isPassthroughMode != null`

Reply:
Out of curiosity -- when will the `refreshToken` be missing _without_ being in passthrough mode? I'm not sure which earlier thread you're referencing (unless it's the one about the `TOKEN_EXPIRED` error). On Android, at least, I've been treating a missing refresh token as an indicator of passthrough mode.

Reply:
Ah, found this line in the old SDK: https://github.com/firebase/firebase-js-sdk/blob/a69220b519185239c364f7a5b3d5983e9c87d7c1/packages/auth/src/token.js#L272

Reply:
@yuchenshi - Yep! Should be fine since isPassthroughMode is false by default

@rosalyntan - Yeah, thanks for linking!